homework-solutions/js/hw-9/task-2.js

```diff
@@ -14,23 +14,41 @@ const characters = [
 ];
 
 function addCharacter(character) {
-  // Ваш код
+  if (typeof character.name !== 'string' || typeof character.age !== 'number') {
+    throw new Error('Type of objects are incorrect');
+  } 
+  characters.push(character);
 }
 
 function getCharacter(name) {
-  // Ваш код
+  return characters.find((user) => user.name === name);
 }
 
 function getCharactersByAge(minAge) {
-  // Ваш код
+  if (typeof minAge !== 'number') {
+    throw new Error('Type of entered age is not a number');
+  } 
+  return characters.filter((user) => user.age >= minAge);
 }
 
 function updateCharacter(name, newCharacter) {
-  // Ваш код
+  const existObj = characters.find((user) => user.name === name);
```

Comment:
Как вариант использовать уже существующий метод
```suggestion
  const existObj = getCharacter(name);
```